src/Microsoft.Restier.Security/InvocationContextExtensions.cs

```diff
@@ -33,11 +33,11 @@ public static class InvocationContextExtensions
         {
             Ensure.NotNull(context, "context");
             Ensure.NotNull(role, "role");
-            var assertedRoles = context.GetProperty<List<string>>(AssertedRoles);
+            var assertedRoles = context.ApiContext.GetProperty<List<string>>(AssertedRoles);
```

Comment:
Temporary workaround: leverage the `PropertyBag` from the `ApiContext` scope to store the `AssertedRoles`. The consequence is the information is now per-request not per-invocation.